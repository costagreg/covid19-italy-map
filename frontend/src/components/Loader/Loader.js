import React, { useState, useEffect } from 'react'
import './Loader.scss'

export default function Loader({ delay = 1000 }) {
  const [isLoading, setIsLoading] = useState(false)

  useEffect(() => {
    const timer = setTimeout(() => {
      setIsLoading(true)
    }, delay)
    return () => clearTimeout(timer)
  }, [])

  return isLoading ? (
    <div className="loader">
      <span title="Loading" className="loader__circle" />
    </div>
  ) : null
}
