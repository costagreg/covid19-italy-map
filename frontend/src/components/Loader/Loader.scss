.loader {
  position: absolute;
  background: #f5f5f5;
  width: 100%;
  height: 100%;

  &__circle {
    position: absolute;
    left:50%;
    top:50%;
    display: inline-block;
    width: 32px;
    height: 32px;

    &:after {
      content: " ";
      display: block;
      width: 32px;
      height: 32px;
      margin: 8px;
      border-radius: 50%;
      border: 6px solid rgb(109, 106, 106);
      border-color: rgb(109, 106, 106)transparent rgb(109, 106, 106) transparent;
      animation: loaderAnimation 1.2s linear infinite;
    }
  }

  @keyframes loaderAnimation {
    0% {
      transform: rotate(0deg);
    }

    100% {
      transform: rotate(360deg);
    }
  }
}