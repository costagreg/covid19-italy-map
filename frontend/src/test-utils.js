const domTestingLib = require('@testing-library/dom')
const { queryHelpers } = domTestingLib

export 