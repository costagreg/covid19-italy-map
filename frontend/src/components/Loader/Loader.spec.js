import { render, waitFor } from '@testing-library/react'
import React from 'react'
import Loader from './Loader'
import { act } from 'react-dom/test-utils'

jest.useFakeTimers()

describe('Loader', () => {
  describe('@render', () => {
    it('renders the DataTable without any error', () => {
      const { asFragment } = render(<Loader />)

      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('@event', () => {
    it('shows loading after X seconds', async () => {
      const { queryByTitle } = render(<Loader delay={10000} />)

      act(() => {
        jest.advanceTimersByTime(1000)
      })

      expect(queryByTitle('Loading')).toBeNull()

      act(() => {
        jest.advanceTimersByTime(10000)
      })

      expect(queryByTitle('Loading')).toBeTruthy()
    })
  })
})
