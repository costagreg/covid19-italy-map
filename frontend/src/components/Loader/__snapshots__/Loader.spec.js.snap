// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader @render renders the DataTable without any error 1`] = `<DocumentFragment />`;
